markdown/events/2021/bytesize-5-dsl2-module-development.md

```diff
@@ -32,3 +32,257 @@ This week, Harshil Patel ([@drpatelh](http://github.com/drpatelh/)) will present
+
+[25:02](https://www.youtube.com/watch?v=ggGGhTMgyHI&t=1502)
+And a real case example is here, where star (which has zillions of parameters possible so I've truncated for good measure there) but you can have all of these parameters provided there.
+
+You can provide one or more of these known types of standardized variables depending on you and the reason for that is because these variables are initialized at the module level and so you won't get errors if they're not initialized via your config.
+
+[25:31](https://www.youtube.com/watch?v=ggGGhTMgyHI&t=1531)
+In this case, we're setting sensible defaults at the module level which means that you don't have to provide a value for all of them. Another example is because it's just a groovy map you can, for example that star_align process, if you have a parameter in the pipeline that specifically needs to be evaluated for you to add another option to that argument you can just do it because it's appending to a string. If you want to publish files, say if you've got a parameter that says save_unaligned files, then you can just put the additional files within this groovy map and then it will publish those as well.
+
+[26:07](https://www.youtube.com/watch?v=ggGGhTMgyHI&t=1567)
+It's quite flexible in the way that you can create and pass these options to the modules and as I mentioned before, you just pass them using this addParams directive.
+
+We have had to write a few functions to customize some behavior and also, again, for simplicity in terms of the way that we're dealing with these files. So, there's only three functions that I've had to write that we are now importing into the module script and these are just custom groovy functions mainly used for publishing files and also passing arguments to these command lines tools. So, in this case, as I mentioned before, you are initializing these options here at the module level if they're not provided by the workflow and so they set sensible defaults.
+
+[26:54](https://www.youtube.com/watch?v=ggGGhTMgyHI&t=1614)
+For example, getSoftwareName here will just use the task.process Nextflow variable and it will just get this bit here. So, by default, this module will publish the files in your output directory and a folder called samtools; which is why in the previous slide I didn't have to provide an output directory for fastqc because I just want the reports to go in a directory called fastqc anyway so i don't need to overwrite that.
+
+[27:20](https://www.youtube.com/watch?v=ggGGhTMgyHI&t=1640)
+Similarly, these, these and these are all again initiated here in this initOptions map which just initiates decent values now. This works, it worked for the rna-seq pipeline, it may not be as comprehensive, it's slightly annoying where some tools require three arguments and so you have `options.args3` which is slightly ugly but it works until we have a better solution.
```

Comment:
```suggestion
Similarly, these, these and these are all again initiated here in this initOptions map which just initiates decent values now. This works, it worked for the rna-seq pipeline, it may not be as comprehensive, it's slightly annoying where some tools require three arguments and so you have options.args3 which is slightly ugly but it works until we have a better solution.
```

Removed for consistency, as not used anywhere else.